anitya/tests/lib/versions/test_rpm.py

```diff
@@ -0,0 +1,226 @@
+        """Assert prereleases are less than non-prereleases."""
+        old_version = rpm.RpmVersion(version='v1.1.0rc1')
+        new_version = rpm.RpmVersion(version='v1.1.0')
+        self.assertTrue(old_version < new_version)
+        self.assertFalse(new_version < old_version)
+
+    def test_lt_both_prerelease(self):
+        """Assert prereleases sort numerically."""
+        old_version = rpm.RpmVersion(version='v1.1.0rc1')
+        new_version = rpm.RpmVersion(version='v1.1.0rc11')
+        self.assertTrue(old_version < new_version)
+        self.assertFalse(new_version < old_version)
+
+    def test_lt_both_prerelease_one_unversioned(self):
+        """Assert unversioned prereleases are less than versioned ones."""
+        old_version = rpm.RpmVersion(version='v1.1.0rc')
+        new_version = rpm.RpmVersion(version='v1.1.0rc1')
+        self.assertTrue(old_version < new_version)
+        self.assertFalse(new_version < old_version)
+
+    def test_le(self):
+        """Assert RpmVersion supports <= comparison."""
+        old_version = rpm.RpmVersion(version='v1.0.0')
+        equally_old_version = rpm.RpmVersion(version='v1.0.0')
+        new_version = rpm.RpmVersion(version='v1.1.0')
+        self.assertTrue(old_version <= new_version)
+        self.assertTrue(old_version <= equally_old_version)
+        self.assertFalse(new_version <= old_version)
+
+    def test_gt(self):
+        """Assert RpmVersion supports > comparison."""
+        old_version = rpm.RpmVersion(version='v1.0.0')
+        new_version = rpm.RpmVersion(version='v1.1.0')
+        self.assertTrue(new_version > old_version)
+        self.assertFalse(old_version > new_version)
+
+    def test_ge(self):
+        """Assert RpmVersion supports >= comparison."""
+        old_version = rpm.RpmVersion(version='v1.0.0')
+        equally_new_version = rpm.RpmVersion(version='v1.1.0')
+        new_version = rpm.RpmVersion(version='v1.1.0')
+        self.assertFalse(old_version >= new_version)
+        self.assertTrue(new_version >= equally_new_version)
+        self.assertTrue(new_version >= old_version)
+
+    def test_eq(self):
+        """Assert RpmVersion supports == comparison."""
+        old_version = rpm.RpmVersion(version='v1.0.0')
+        new_version = rpm.RpmVersion(version='v1.0.0')
+        self.assertTrue(new_version == old_version)
+
+    def test_eq_both_unnumbered_prereleases(self):
+        """Assert two prereleases of the same type without versions are equal."""
+        old_version = rpm.RpmVersion(version='1.0.0beta')
+        new_version = rpm.RpmVersion(version='1.0.0beta')
+        self.assertTrue(new_version == old_version)
+
+    def test_eq_one_v_prefix(self):
+        """Assert versions that are the same except one has a v prefix are equal."""
+        old_version = rpm.RpmVersion(version='v1.0.0-1.fc26')
+        new_version = rpm.RpmVersion(version='1.0.0-1.fc26')
+        self.assertTrue(new_version == old_version)
```

Comment:
Huh, if these still pass when the system RPM bindings aren't available, I guess my fallback emulation code actually works as advertised and I can remove that caveat from the Stack Overflow answer :)